### PR TITLE
#14634: Remove usage of ARCH_NAME sp constants MEM_L1_SIZE

### LINKED ISSUE #14634: command_queue_interface.hpp uses ARCH_NAME specific include path to access MEM_L1_SIZE, MEM_ETH_SIZE
https://github.com/tenstorrent/tt-metal/blob/97731c4e64db1306a751ddc467e6e1e65f1009f7/tt_metal/impl/dispatch/command_queue_interface.hpp#L195

These defines come from `dev_mem_map.h`. Need to get them at runtime from hal.

### PR DESCRIPTION
**Pull request by blozano-tt**
### Ticket
Closes #14634 

### Problem description
MEM_L1_BASE, MEM_L1_SIZE
MEM_ETH_BASE, MEM_ETH_SIZE
All come from ARCH_NAME specific includes.

Can't use them if we want a singular build.

### What's changed
Use HAL to get constants

### Checklist
https://github.com/tenstorrent/tt-metal/actions/runs/11850613530
- [x] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

**Review by tt-aho — COMMENTED** [review]
I'm not really involved with HAL discussion/design, but why do we not want the mem size to be behind the hal?

@pgkeller @abhullar-tt

**Comment by blozano-tt** [comment]
> I'm not really involved with HAL discussion/design, but why do we not want the mem size to be behind the hal?
> 
> @pgkeller @abhullar-tt

@tt-aho 

We can add it if desired. I avoided adding it, because every parameter we put behind Hal becomes scrutinized ... "should we / should we not / is the current usage dead code?".

Since I could infer the size from what already exists in Hal, I went that route.

If you all think its simpler or smarter to add it to Hal, I'm happy to adjust.

File: tt_metal/impl/debug/sanitize_noc_host.hpp [hunk 1]
```diff
@@ -2,15 +2,38 @@
...
+           hal.get_dev_addr(core_type, tt::tt_metal::HalL1MemAddrType::UNRESERVED);
+  }
+  inline std::uint32_t mem_l1_size(const tt::tt_metal::Hal& hal) {
+    return compute_size(hal, tt::tt_metal::HalProgrammableCoreType::TENSIX);
+  }
+  inline std::uint32_t mem_eth_size(const tt::tt_metal::Hal& hal) {
+    if (hal.get_arch() == tt::ARCH::GRAYSKULL) {
+      return 0;
+    }
+    return compute_size(hal, tt::tt_metal::HalProgrammableCoreType::IDLE_ETH);
```
Comment (pgkeller) [inline tt_metal/impl/debug/sanitize_noc_host.hpp:29]: the compute size should be behind the hal 100%, ie, hal should return 0 for GS

File: tt_metal/impl/debug/sanitize_noc_host.hpp [hunk 2]
```diff
@@ -2,15 +2,38 @@
...
 
+#include <cstdint>
+
 #include "noc/noc_parameters.h"
 #include "noc/noc_overlay_parameters.h"
 
 #pragma once
 
+#undef MEM_L1_BASE
+#undef MEM_ETH_BASE
```
Comment (pgkeller) [inline tt_metal/impl/debug/sanitize_noc_host.hpp:13]: don't understand this 100%

File: tt_metal/impl/dispatch/command_queue_interface.hpp [hunk 3]
```diff
@@ -15,6 +15,23 @@
 
 #include <magic_enum.hpp>
 
+namespace {
```
Comment (pgkeller) [inline tt_metal/impl/dispatch/command_queue_interface.hpp:18]: why duplicated?  this could live at the llrt level (basically a low level utility)

**Review by pgkeller — CHANGES_REQUESTED** [review]

File: tt_metal/impl/debug/sanitize_noc_host.hpp (hunk 1 above)
Comment (blozano-tt) [reply tt_metal/impl/debug/sanitize_noc_host.hpp]: Thanks for the feedback, moved to hal now.

File: tt_metal/llrt/hal.hpp [hunk 4]
```diff
@@ -263,3 +264,9 @@ extern Hal hal;
 
 }  // namespace tt_metal
 }  // namespace tt
+
+
+#define HAL_MEM_L1_BASE tt::tt_metal::hal.get_dev_size(tt::tt_metal::HalProgrammableCoreType::TENSIX, tt::tt_metal::HalL1MemAddrType::BASE)
+#define HAL_MEM_L1_SIZE tt::tt_metal::hal.get_dev_addr(tt::tt_metal::HalProgrammableCoreType::TENSIX, tt::tt_metal::HalL1MemAddrType::BASE)
+#define HAL_MEM_ETH_BASE tt::tt_metal::hal.get_dev_size(tt::tt_metal::HalProgrammableCoreType::IDLE_ETH, tt::tt_metal::HalL1MemAddrType::BASE)
+#define HAL_MEM_ETH_SIZE tt::tt_metal::hal.get_dev_addr(tt::tt_metal::HalProgrammableCoreType::IDLE_ETH, tt::tt_metal::HalL1MemAddrType::BASE)
```
Comment (blozano-tt) [inline tt_metal/llrt/hal.hpp:272]:
```suggestion
#define HAL_MEM_L1_BASE tt::tt_metal::hal.get_dev_addr(tt::tt_metal::HalProgrammableCoreType::TENSIX, tt::tt_metal::HalL1MemAddrType::BASE)
#define HAL_MEM_L1_SIZE tt::tt_metal::hal.get_dev_sizett::tt_metal::HalProgrammableCoreType::TENSIX, tt::tt_metal::HalL1MemAddrType::BASE)
#define HAL_MEM_ETH_BASE tt::tt_metal::hal.get_dev_addr(tt::tt_metal::HalProgrammableCoreType::IDLE_ETH, tt::tt_metal::HalL1MemAddrType::BASE)
#define HAL_MEM_ETH_SIZE tt::tt_metal::hal.get_dev_size(tt::tt_metal::HalProgrammableCoreType::IDLE_ETH, tt::tt_metal::HalL1MemAddrType::BASE)
```

**Review by pgkeller — APPROVED** [review]

**Pull request merged**